src/screens/Register/RegisterScreen.tsx

```diff
@@ -41,11 +41,13 @@ const RegisterScreen: React.FC<RegisterScreenProps> = ({ setCurrentUser }) => {
       (async () =>
         // set access token
         await SecureStore.setItem(SecureStoreEnum.TOKEN, signup.accessToken))();
-      setCurrentUser(signup.user.username);
+      setCurrentUser({ username: signup.user.username, id: signup.user.id });
       navigateToHome();
     } else if (error) {
-      (async () =>
-        await SecureStore.deleteItem(SecureStoreEnum.TOKEN).catch())();
+      async () => {
```

Comment:
This doesn't get executed. It's just declaring an async function. It needs to be an immediately invoked function. It needs the wrapper brackets